### PR TITLE
Get CI build working again

### LINKED ISSUE #136: Fix CI on master
Currently, we can't maintain a 'light touch' maintenance presence on this project because the CI we use to validate changes is broken. The integration tests take too long to run so Travis CI times-out. We should either parallelize our tests or move to a CI provider that is more forgiving to long run-times.

### PR DESCRIPTION
**Pull request by cstroe**
Fix #136 

Major fixes here:
* added template support for the test fixture project files (using [FreeMarker](https://freemarker.apache.org/))
* get as many of the Play 2.3 tests working as we can, skip the rest (mostly tests that don't apply to Play 2.3)
* split up the Travis CI build into one build per version + documentation build (avoids timeout, [example](https://travis-ci.org/github/cstroe/playframework/builds/726509447)) (this might also fix https://github.com/gradle/playframework/issues/75)
* be very verbose about logging test execution, failures, stacktraces

### CODE REVIEW

**Comment by seyashizuo** [comment]
Cool! Waiting for this.

File: src/integTestFixtures/groovy/org/gradle/playframework/fixtures/app/WithFailingTestsApp.groovy [hunk 1]
```diff
@@ -12,7 +12,7 @@ class WithFailingTestsApp extends PlayApp {
 
     @Override
     SourceFile getGradleBuild() {
-        def gradleBuild = sourceFile("", "build.gradle", "basicplayapp")
+        def gradleBuild = sourceFile("", "build.gradle.ftl", "basicplayapp")
```
Comment (JLLeitschuh) [inline src/integTestFixtures/groovy/org/gradle/playframework/fixtures/app/WithFailingTestsApp.groovy:15]: Can you explain why this is named with the extension `.ftl`?

Comment (JLLeitschuh) [reply src/integTestFixtures/groovy/org/gradle/playframework/fixtures/app/WithFailingTestsApp.groovy:15]: NVM, I understand, you're using freemarker.

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/build.gradle.ftl [hunk 2]
```diff
@@ -6,3 +7,10 @@ dependencies {
     implementation "com.typesafe.play:play-guice_2.12:2.6.15"
     implementation "ch.qos.logback:logback-classic:1.2.3"
 }
+
+<#else>
+plugins {
+    id 'org.gradle.playframework'
+}
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/build.gradle.ftl:16]: EOL @ EOF
```suggestion
</#if>

```

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/primary/app/controllers/Application.scala.ftl [hunk 3]
```diff
@@ -18,3 +19,23 @@ class Application @Inject() extends InjectedController {
...
+    Ok(Util.fullStop("Your new application is ready"))
+  }
+
+  def shutdown = Action {
+    System.exit(0)
+    Ok("shutdown")
+  }
+}
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/primary/app/controllers/Application.scala.ftl:41]: EOL @ EOF
```suggestion
</#if>

```

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/primary/build.gradle.ftl [hunk 4]
```diff
@@ -8,3 +9,15 @@ dependencies {
...
+plugins {
+    id 'org.gradle.playframework'
+}
+
+dependencies {
+    implementation project(":submodule")
+    implementation project(":javalibrary")
+}
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/primary/build.gradle.ftl:23]: EOL @ EOF
```suggestion
</#if>

```

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/primary/conf/application.conf.ftl [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+# Root logger:
+logger.root=ERROR
+
+# Logger used by the framework:
+logger.play=INFO
+
+# Logger provided to your application:
+logger.application=DEBUG
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/primary/conf/application.conf.ftl:18]: EOL @ EOF
```suggestion
</#if>
```

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/primary/conf/routes.ftl [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+GET     /submodule                 @controllers.submodule.Application.index
+GET     /assets/*file              @controllers.Assets.at(path="/public", file)
+
+<#else>
+GET     /                          controllers.Application.index
+GET     /shutdown                  controllers.Application.shutdown
+GET     /submodule                 controllers.submodule.Application.index
+GET     /assets/*file              controllers.Assets.at(path="/public", file)
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/primary/conf/routes.ftl:13]: EOL @ EOF
```suggestion
</#if>

```

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/submodule/app/controllers/submodule/Application.scala.ftl [hunk 7]
```diff
@@ -12,3 +13,16 @@ class Application @Inject() extends InjectedController {
...
+
+object Application extends Controller {
+
+  def index = Action {
+    Ok("Submodule page")
+  }
+
+}
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/submodule/app/controllers/submodule/Application.scala.ftl:28]: EOL @ EOF
```suggestion
</#if>

```

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/submodule/build.gradle.ftl [hunk 8]
```diff
@@ -1,9 +1,16 @@
...
     implementation "com.typesafe.play:play-guice_2.12:2.6.15"
     implementation "ch.qos.logback:logback-classic:1.2.3"
 }
+
+<#else>
+plugins {
+    id 'org.gradle.playframework'
+}
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playmultiproject/submodule/build.gradle.ftl:16]: EOL @ EOF
```suggestion
</#if>

```

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/shared/conf/application.conf.ftl [hunk 9]
```diff
@@ -8,4 +12,6 @@ logger.root=ERROR
 logger.play=INFO
 
 # Logger provided to your application:
-logger.application=DEBUG
\ No newline at end of file
+logger.application=DEBUG
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/shared/conf/application.conf.ftl:17]: EOL @ EOF
```suggestion
</#if>

```

**Review by JLLeitschuh — CHANGES_REQUESTED** [review]
This looks absolutely amazing. I'm really looking forward to merging this. Thank you so much for putting in the time to fix this.

Please take a quick pass at my minor reviews and fix them up. Once resolved, I'm absolutely looking forward to merging this.

File: src/integTestFixtures/groovy/org/gradle/playframework/fixtures/app/WithFailingTestsApp.groovy (hunk 1 above)
Comment (cstroe) [reply src/integTestFixtures/groovy/org/gradle/playframework/fixtures/app/WithFailingTestsApp.groovy:15]: Yep, `.ftl` is one of the recommended file extensions for a FreeMarker template.  It doesn't really buy much with the IDE, but the test fixtures code checks for the extension to know if it should run the file through the FreeMarker engine.

**Comment by cstroe** [comment]
Thank you very much for the fast review!  I applied your suggestions, but there was one suggestion that got left over (it might have been double suggested because it says there are no changes in the suggestion).  For this I had to request a re-review.

Let me know if there's anything else required.  Also, the build got triggered again, should be a good test of the CI configuration.

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playappwithdependencies/conf/routes.ftl [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+# Routes
+# Home page
+GET     /                           controllers.Application.index
+
+GET     /shutdown                   controllers.Application.shutdown
+
+# Map static resources from the /public folder to the /assets URL path
+GET     /assets/*file               controllers.Assets.at(path="/public", file)
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playappwithdependencies/conf/routes.ftl:21]: EOL @ EOF
```suggestion
</#if>

```

File: src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playcompositebuild/conf/routes.ftl [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+GET     /                          @controllers.Application.index
+GET     /shutdown                  @controllers.Application.shutdown
+GET     /assets/*file              @controllers.Assets.at(path="/public", file)
+
+<#else>
+GET     /                          controllers.Application.index
+GET     /shutdown                  controllers.Application.shutdown
+GET     /assets/*file              controllers.Assets.at(path="/public", file)
+
+</#if>
```
Comment (JLLeitschuh) [inline src/integTestFixtures/resources/org/gradle/playframework/fixtures/app/playcompositebuild/conf/routes.ftl:11]: EOL @ EOF
```suggestion
</#if>

```

**Review by JLLeitschuh — CHANGES_REQUESTED** [review]
There are a few additional EOL @ EOF issues.

Can you make sure that every file that you've contributed has an End of Line (EO) (IE. the newline character) @ EOF (IE the last line in the file).

**Comment by cstroe** [comment]
Sorry about that, I think what happened is when I ran the commands for the DCO signing after accepting the changes on GitHub, I didn't pull locally, so overwrite all the newline additions changes.

I double checked the newline additions, they should be there now.

**Review by JLLeitschuh — APPROVED** [review]
Looks good to me!!

**Pull request merged**